tests/test_hub.py

```diff
@@ -73,16 +63,18 @@ def uploaded_bucket_file(result_bucket_name, api_client, rng):
 def test_link_bucket_file_to_analysis(uploaded_bucket_file, analysis_id, api_client):
     _, bucket_file = uploaded_bucket_file
 
+    analysis_bucket = api_client.get_analysis_bucket(analysis_id, "RESULT")
+
     # check that the analysis file was created
     analysis_file = api_client.link_bucket_file_to_analysis(
-        analysis_id, bucket_file.id, bucket_file.name, bucket_type="RESULT"
+        analysis_bucket.external_id, bucket_file.id, bucket_file.name
```

Comment:
Might need some clarification on what the new `/analysis-bucket-files` endpoint *really* does. It seems as if the parameter order is off but flipping them doesn't fix it. And with my current understanding, `external_id` and `bucket_id` refer to the same thing. Where is the link to the analysis?

Reply:
The analysis-bucket-files endpoint is not a new endpoint. The endpoint /analysis-files has been renamed to the endpoint /analysis-bucket-files. The external_id refers to the identifier in the storage service.
The endpoint /analysis-buckets has been added to store a specific analysis and a type (temp, result, code), as well as the external_id (bucket identifier in the storage service)